purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1497,15 +1463,21 @@ class Purchases internal constructor(
             override fun onPurchasesFailedToUpdate(purchasesError: PurchasesError) {
                 synchronized(this@Purchases) {
                     getAndClearProductChangeCallback()?.dispatch(purchasesError)
-                        ?: state.purchaseCallbacksByProductId.let { purchaseCallbacks ->
-                            state = state.copy(purchaseCallbacksByProductId = emptyMap())
-                            purchaseCallbacks.values.forEach { it.dispatch(purchasesError) }
-                        }
+                        ?: errorAllPurchaseCallbacks(purchasesError)
```

Comment:
It looks like `errorAllPurchaseCallbacks` also has a `synchronized(this@Purchases)` in it... will this cause any issues since this is being called from within a `synchronized(this@Purchases)` already?

Reply:
i don't _think _ so... @vegaro mind confirming?

Reply:
it's ok, synchronized is re-entrant in Java so it won't cause a deadlock like in other languages